packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+/**
+ * React hook that measures virtualized space based on a static size to ensure optimized virtualization length.
+ */
+export const useStaticVirtualizerMeasure = (
+  defaultItemSize: number,
+  scrollView: HTMLElement | null = null,
+  direction: 'vertical' | 'horizontal' = 'vertical',
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+} => {
+  const [virtualizerLength, setVirtualizerLength] = React.useState(0);
+  const [virtualizerBufferItems, setVirtualizerBufferItems] = React.useState(0);
+  const [virtualizerBufferSize, setVirtualizerBufferSize] = React.useState(0);
+
+  const { targetDocument } = useFluent();
+
+  // If no ref is provided, the virtualizer should be in the main body document's scroll.
+  const _scrollView = scrollView ?? targetDocument?.documentElement ?? null;
+
+  const container = React.useRef<HTMLElement | null>(null);
+
+  // the handler for resize observer
+  const handleResize = React.useCallback(() => {
+    const containerSize =
+      direction === 'vertical'
+        ? container.current?.getBoundingClientRect().height
+        : container.current?.getBoundingClientRect().width;
+
+    if (!containerSize) {
```

Comment:
this edge case where the ref is not set could be eliminated quite easily if the hook returns a ref

Reply:
Went for a ref callback function here to ensure we initialize on mount immediately after setting, defaults to document.body to ensure no null initialization so we can bypass this check.

Reply:
What you implemented was not what I meant. refs shouldn't be used as arguments, rather hooks should return refs. so in this case I would expect the usage pattern of `useVirtualizerScrollRef` to be something like

```tsx 
const scrollRef = useVirtualizerScrollRef()

return <div ref={scrollRef} />
```

Reply:
Updated with a proper ref callback